### PR TITLE
feat: #438 admin search by fom number

### PR DESCRIPTION
**Pull request by ianliuwk1019**
ref: #438 
- label change from FOM ID to FOM Number.
- add FOM Number search input for admin and api.
- update swagger and api-client.

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
 - [api](https://fom-12.apps.silver.devops.gov.bc.ca/api)
 - [admin](https://fom-12.apps.silver.devops.gov.bc.ca/admin)
 - [public](https://fom-12.apps.silver.devops.gov.bc.ca/public)

Once merged, code will be promoted and handed off to following workflow run.
 - [Main Merge Workflow](https://github.com/bcgov/nr-fom/actions/workflows/merge-main.yml)

---

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
 - [api](https://fom-12.apps.silver.devops.gov.bc.ca/api)
 - [admin](https://fom-12.apps.silver.devops.gov.bc.ca/admin)
 - [public](https://fom-12.apps.silver.devops.gov.bc.ca/public)

Once merged, code will be promoted and handed off to following workflow run.
 - [Main Merge Workflow](https://github.com/bcgov/nr-fom/actions/workflows/merge-main.yml)

### CODE REVIEW

File: admin/src/app/search/search.component.html [hunk 1]
```diff
@@ -25,13 +25,27 @@ <h1 class="text-muted">Search FOMs</h1>
...
+        </div>
+
+        <div class="form-group col-md-2">
           <label for="fspIdInput" style="white-space: nowrap;">
                 FSP ID
           </label>
           <input id="fspIdInput" name="fFspId" type="number"
                 min="0" step="1"
-                oninput="javascript: if (this.value.length > this.maxLength) this.value = this.value.slice(0, 9);"
+                oninput="this.value = this.value.slice(0, 9)"
```
Comment (ianliuwk1019) [inline admin/src/app/search/search.component.html:48]: It works previously, for restricting length to 9 digits; I only removed unnecessary 'javascript' part and condition.
New FOM Number field also follows this one.
I could not find a better way to be close to Angular way for event handling and assigning new value to view value in this particular case. This is probably best I can find so far.
The only case it seems not working is entering multiple '0'(s), but I feel we can ignore this problem for now...

File: api/src/app/modules/project/project.service.ts [hunk 2]
```diff
@@ -37,10 +37,10 @@ export class ProjectFindCriteria {
   projectId?: number;
 
   applyFindCriteria(query: SelectQueryBuilder<Project>) {
-    if (this.projectId) {
+    if (this.projectId >= 0) {
```
Comment (ianliuwk1019) [inline api/src/app/modules/project/project.service.ts:40]: Fix case when value is 0.

File: admin/src/app/search/search.component.html (hunk 1 above)
Comment (MCatherine1994) [reply admin/src/app/search/search.component.html:48]: Hi Ian, I'm not sure if I understand correctly, but when I put a number longer than 11 chars, I got an error
Screen Shot 2024-07-30 at 10 25 29 AM

Comment (ianliuwk1019) [reply admin/src/app/search/search.component.html:48]: Interesting, maybe I only verify locally, thanks, let me check on why it is allowing for more digits.

Comment (ianliuwk1019) [reply admin/src/app/search/search.component.html:48]: Hmm....this is strange, locally it does not allow for more than 9 digits, but for deployed app, it does not work. And verify in TEST (which new code is not there yet) and it does have this bug. So it is an existing bug that did not notice locally.

Comment (MCatherine1994) [reply admin/src/app/search/search.component.html:48]: I see, maybe we could create another ticket to check this one?

**Review by MCatherine1994 — APPROVED** [review]

**Review by basilv — APPROVED** [review]

File: admin/src/app/search/search.component.html (hunk 1 above)
Comment (ianliuwk1019) [reply admin/src/app/search/search.component.html:48]: Spent a little time, can't seem to make `<input type="number">` to restrict digits with event listener to work.
I seem to recall Basil mentioned before to use `<input type="text">` is better (however, I do see some minor drawback using 'text' for integer field) but can work.
Will ignore this for now and create new ticket to deal with it (similar like public side or using some sort of Angular directive for better input experience).

**Pull request merged**